File: $_algotsu-kaizen/problemo-set/longest-binary-subsequence-less-than-or-equal-to-k.c

int longestSubsequence(char* s, int k) {
    register int n = 0x0, result = 0x0, i, bitIDX;
    for (i = strlen(s) - 0x1, bitIDX = 0x0; i >= 0x0; --i, ++bitIDX)
        (s[i] == '0' || bitIDX < 0x1F && (n += 0x1 << bitIDX) <= k) && ++result;
    return result;
}

File: $_algotsu-kaizen/problemo-set/longest-binary-subsequence-less-than-or-equal-to-k.rs
impl Solution {

    pub fn longest_subsequence(string: String, k: i32) -> i32 {
        let (mut result, mut n): (i32, i64) = (0x0 , 0x0);

        for (bit_idx, bit) in string.chars().rev().enumerate() {
            if bit == '0' {
                result += 1;
            } else if bit_idx < 0x1F {
                n += 0x1 << bit_idx;
                if n <= k as i64 {
                    result += 0x1
                }
            }
        }

        result
    }
}
